modules/alloy/src/main/ts/ephox/alloy/behaviour/docking/DockingApis.ts

```diff
@@ -44,39 +44,77 @@ const updateVisibility = (component: AlloyComponent, config: DockingConfig, stat
+  component: AlloyComponent,
+  config: DockingConfig,
+  state: DockingState,
+  viewport: DockingViewport,
+  morph: Dockables.FixedMorph
+) => {
+  // This "updateVisibility" call is potentially duplicated with the
+  // call in refreshInternal for isDocked. We might want to consolidate them.
+  // The difference between them is the "morphToDocked" flag.
+  updateVisibility(component, config, state, viewport, true);
+  morphToCoord(component, config, state, morph.positionCss);
+};
+
+const applyMorph = (
+  component: AlloyComponent,
+  config: DockingConfig,
+  state: DockingState,
+  viewport: DockingViewport,
+  morph: Dockables.MorphInfo
+) => {
+  // Apply the morph result depending on its type
+  switch (morph.morph) {
+    case 'static': {
+      return morphToStatic(component, config, state);
+    }
+    case 'absolute': {
+      return morphToCoord(component, config, state, morph.positionCss);
+    }
+    case 'fixed': {
+      return applyFixedMorph(component, config, state, viewport, morph);
+    }
+  }
+};
+
 const refreshInternal = (component: AlloyComponent, config: DockingConfig, state: DockingState): void => {
   // Absolute coordinates (considers scroll)
-  const viewport = config.lazyViewport(component);
+  const viewport: DockingViewport = config.lazyViewport(component);
   // If docked then check if we need to hide/show the component
   const isDocked = state.isDocked();
   if (isDocked) {
     updateVisibility(component, config, state, viewport);
   }
 
   Dockables.getMorph(component, viewport, state).each((morph) => {
-    // Apply the morph result
-    morph.fold(
-      () => morphToStatic(component, config, state),
-      (position) => morphToCoord(component, config, state, position),
-      (position) => {
-        updateVisibility(component, config, state, viewport, true);
-        morphToCoord(component, config, state, position);
-      }
-    );
+    applyMorph(component, config, state, viewport, morph);
   });
 };
 
 const resetInternal = (component: AlloyComponent, config: DockingConfig, state: DockingState): void => {
   // Morph back to the original position
   const elem = component.element;
   state.setDocked(false);
-  Dockables.getMorphToOriginal(component, state).each((morph) => {
-    morph.fold(
-      () => morphToStatic(component, config, state),
-      (position) => morphToCoord(component, config, state, position),
-      Fun.noop
-    );
-  });
+  const viewport = config.lazyViewport(component);
+  Dockables.getMorphToOriginal(component, viewport, state).each(
+    (staticOrAbsoluteMorph: Dockables.StaticMorph | Dockables.AbsoluteMorph) => {
+      // This code is very similar to the "applyMorph" function above. The main difference
+      // is that it doesn't consider fixed position, because something that is docking
+      // can't currently start with fixed position
+      switch (staticOrAbsoluteMorph.morph) {
+        case 'static': {
+          return morphToStatic(component, config, state);
```

Comment:
This looks weird it looks like it's returning something but since the function is for each maybe just make it two lines so it's more explicit that it's a void return or use break.

Reply:
Can do

- [ ] make it more explicit this is void return type.